scripts/prepare_dutchf3.py

```diff
@@ -105,66 +128,118 @@ def split_patch_train_val(data_dir, stride, patch_size, per_val=0.2, log_config=
     logger.info("Splitting data into patches .... ")
     logger.info(f"Reading data from {data_dir}")
 
-    labels_path = _get_labels_path(data_dir)
-    logger.info(f"Loading {labels_path}")
-    labels = np.load(labels_path)
+    logger.info(f"Loading {label_file}")
+    labels = np.load(label_file)
     logger.debug(f"Data shape [iline|xline|depth] {labels.shape}")
 
     iline, xline, depth = labels.shape
     # Inline sections
-    train_iline_range, test_iline_range = _get_aline_range(iline, per_val)
+    train_iline_range, val_iline_range = _get_aline_range(iline,
+                                                           per_val,
+                                                           slice_steps)
 
     # Xline sections
-    train_xline_range, test_xline_range = _get_aline_range(xline, per_val)
+    train_xline_range, val_xline_range = _get_aline_range(xline,
+                                                           per_val,
+                                                           slice_steps)
 
     # Generate patches from sections
-    # Process inlines
-    horz_locations = range(0, xline - patch_size + 1, stride)
-    vert_locations = range(0, depth - patch_size + 1, stride)
-    logger.debug("Generating Inline patches")
-    logger.debug(horz_locations)
+    # Vertical locations is common to all patches processed
+    vert_locations = range(0, depth - patch_size, stride)
     logger.debug(vert_locations)
 
+    # Process inlines
+    # iline = xline x depth
+    val_iline = math.floor(xline * per_val / 2)
```

Comment:
why do we divide by 2 here? is it because the val sections are taken from both ends of the volume?

Reply:
yes, we split half to each edge of the volume.